examples/long-short/long-short.go

```diff
@@ -0,0 +1,387 @@
+package main
+
+import (
+	"fmt"
+	"math"
+	"os"
+	"sort"
+	"time"
+
+	"github.com/alpacahq/alpaca-trade-api-go/alpaca"
+	"github.com/alpacahq/alpaca-trade-api-go/common"
+	"github.com/shopspring/decimal"
+)
+
+type alpacaClientContainer struct {
+	client    *alpaca.Client
+	long      bucket
+	short     bucket
+	allStocks []stockField
+	blacklist []string
+}
+type bucket struct {
+	bucketType  string
+	list        []string
+	qty         int
+	adjustedQty int
+	equityAmt   float64
+}
+type stockField struct {
+	name string
+	pc   float64
+}
+
+var alpacaClient alpacaClientContainer
+
+func init() {
+	API_KEY := "YOUR_API_KEY_HERE"
+	API_SECRET := "YOUR_API_SECRET_HERE"
+	BASE_URL := "https://paper-api.alpaca.markets"
+
+	os.Setenv(common.EnvApiKeyID, API_KEY)
+	os.Setenv(common.EnvApiSecretKey, API_SECRET)
+	alpaca.SetBaseUrl(BASE_URL)
+
+	// Format the allStocks variable for use in the class.
+	allStocks := []stockField{}
+	stockList := []string{"DOMO", "TLRY", "SQ", "MRO", "AAPL", "GM", "SNAP", "SHOP", "SPLK", "BA", "AMZN", "SUI", "SUN", "TSLA", "CGC", "SPWR", "NIO", "CAT", "MSFT", "PANW", "OKTA", "TWTR", "TM", "RTN", "ATVI", "GS", "BAC", "MS", "TWLO", "QCOM"}
+	for _, stock := range stockList {
+		allStocks = append(allStocks, stockField{stock, 0})
+	}
+
+	alpacaClient = alpacaClientContainer{
+		alpaca.NewClient(common.Credentials()),
+		bucket{"Long", []string{}, -1, -1, 0},
+		bucket{"Short", []string{}, -1, -1, 0},
+		make([]stockField, len(allStocks)),
+		[]string{},
+	}
+
+	copy(alpacaClient.allStocks, allStocks)
+}
+
+func main() {
+	// First, cancel any existing orders so they don't impact our buying power.
+	status, until, limit := "open", time.Now(), 100
+	orders, _ := alpacaClient.client.ListOrders(&status, &until, &limit)
+	for _, order := range orders {
+		_ = alpacaClient.client.CancelOrder(order.ID)
+	}
+
+	// Wait for market to open.
+	fmt.Println("Waiting for market to open...")
+	for {
+		isOpen := alpacaClient.awaitMarketOpen()
+		if isOpen {
+			break
+		}
+		time.Sleep(2000 * time.Millisecond)
```

Comment:
Any reason for this to be 2000 rather than 1000?

Reply:
Should actually be 60000 to correspond to a minute, forgot to change it